meetings/2021-06-16.md

```diff
@@ -0,0 +1,151 @@
+
+Justin: the interop spec just states: "apply permissions". A working, deployed private implementation with custom code (not using UMA, just secure redirects).
+
+Justin: UMA is an implementation of this pattern with good adoption. Designed to work with the OAuth2 stack and complements some of the other standards we are using in Solid.
+
+Justin: this is something we can look at as a trusted pattern. Not saying "UMA or bust", but thinks we should look at the problems it can help us with.
+
+Henry: What I found is that we have two things doing Authorization: the guard on the server, and this other Authorization server. This seems confusing. Why do we need two authorization systems?
+
+
+On the other hand thinking about the [launcher app](https://github.com/solid/authorization-panel/blob/main/proposals/LauncherApp.md) I proposed a couple of years ago, I could see how there are two Authorization roles:
+  * the Guard on the server, authorizing access to the resource
+  * the Guard near the wallet (essentially the Launcher App) authorizing when to sign headers, give out tokens or provide credentials
+  
+Henry: on the one side, you may have a wallet; on the other you verify the credential. Is that what UMA is trying to do? 
+
+Aaron: First of all if you have an authorization server like UMA (just a specialisation of OAuth 2), you can actually dispense the authorization guard on your AS. It leads to a much cleaner separation of concern. All you need is that the token would be valid for one specific request.
+
+Henry: I am very skeptical about this. There are reasons to be skeptical, as OAuth hardly works for our use case of decentralised hyper apps fetching data by following links. (In fact it seems to me more like  a catastrophe, a point that was echoed by some in last week's discussion on the Credentials CG). This is requiring me to have to read UMA and analyse why it might be wrong or if it is right somehow. That's a lot of work.
+
+Aaron: I think you've clearly articulated your position, thank you. OAuth is a very general authorization framework that has worked quite successfully and is widely deployed on the web today.
+
+Henry: It doesn't solve the problem we're trying to address. It solves the problem of Facebook asking people to enter their gmail password to allow FB to contact their friends and invite them to their social network.
+
+Aaron: @bblfish:matrix.org you are conflating OIDC and OAuth2. We're talking about OAuth2, not OIDC.
+
+Henry: one does not need in depth knowledge of either technology to understand its limitations, which arise directly out of not accepting the Semantic web or linked data principles. By deliberately ignoring these technologies, they will have had to come up with very awkward answers to problems that would have had clean mathematical/logical solutions otherwise.
```

Comment:
```suggestion
Henry: one does not need in depth knowledge of either technology to understand their limitations, which arise directly out of not accepting the Semantic web or linked data principles. By deliberately ignoring these technologies, they have had to come up with very awkward answers to problems that would have had clean mathematical/logical solutions otherwise.
```